### PR TITLE
Fixes crash when resource file is corrupted

### LINKED ISSUE #32511: .vs Visual Script file corrupting to RSCC
**Godot version:**
3.1.1 and 3.2

**OS/device including version:**
Windows 10 and Mac OS 10.14.6

**Issue description:**
Visual Script files are becoming corrupted and the data disappears. This is bad because you lose your entire visual graph and there is no way to get it back. What happens is that when you open up one of the corrupted files a blank .vs file opens at the top of the list and the file itself opens in the code editor and just says "RSCC."

**Steps to reproduce:**
I am not sure exactly how to get the file to this state, but I have attached a minimal project below where there are 2 .vs files that are 1 version off of each other. I use Dropbox and so I can go back every time the file changes. One of the files doesn't corrupt (the one that has (1) at the end) and the other file will corrupt when you open up the file in GODOT. These files are 1 change apart from each other. So what ever I did to corrupt them I did right between. The next file (which I didn't include) is literally just RSCC with no other data in the file. Those 3 files all happened within minutes of each other.

I haven't been able to figure out exactly what causes the file to corrupt, but maybe someone smarter than me can see the difference between the two files and figure out what happened.

**Minimal reproduction project:** 
[InfiniteRunner copy.zip](https://github.com/godotengine/godot/files/3684303/InfiniteRunner.copy.zip)

### PR DESCRIPTION
**Pull request by timothyqiu**
When playing around, I somehow corrupted a Visual Script file. After that, the editor crashes when trying to open the project.

A test project with the corrupted file: [Corrupt.zip](https://github.com/godotengine/godot/files/4184035/Corrupt.zip)

*Bugsquad edit:* Fixes #32511.

### CODE REVIEW

File: core/io/resource_format_binary.cpp [hunk 1]
```diff
@@ -838,7 +838,8 @@ void ResourceInteractiveLoaderBinary::open(FileAccess *p_f) {
 	if (header[0] == 'R' && header[1] == 'S' && header[2] == 'C' && header[3] == 'C') {
 		//compressed
 		FileAccessCompressed *fac = memnew(FileAccessCompressed);
-		fac->open_after_magic(f);
+		error = fac->open_after_magic(f);
+		ERR_FAIL_COND_MSG(error != OK, "Failed to open binary resource file: " + local_path + ".");
```
Comment (akien-mga) [inline core/io/resource_format_binary.cpp]: Aren't we going to leak `f` and `fac`?
I tried to solve this too a few weeks ago, attempting to properly free memory, and I was getting weird crashes :/

Comment (akien-mga) [reply core/io/resource_format_binary.cpp]: See https://github.com/godotengine/godot/issues/32511#issuecomment-574062333

Comment (timothyqiu) [reply core/io/resource_format_binary.cpp]: Ah, yes, it will leak `fac`. I'll look further into the handling of `f`.

**Comment by timothyqiu** [comment]
The additional crash is caused by deleting the same pointer twice. Because `open_after_magic` took over the argument `FileAccess` object, and deleted it in the destructor.

We have functions like `void ResourceInteractiveLoaderBinary::open(FileAccess *p_f)` using `open_after_magic`. They have no way to tell their caller to invalidate the parameter pointer. So, I think it's better to let `open_after_magic`'s caller to decide what to do with the pointer on failure.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.1.

**Pull request merged**